core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java

```diff
@@ -254,8 +254,11 @@ public boolean dropTable(TableIdentifier identifier, boolean purge) {
           // Since the data files and the metadata files may store in different locations,
           // so it has to call dropTableData to force delete the data file.
           CatalogUtil.dropTableData(ops.io(), lastMetadata);
+          return fs.delete(tablePath, true /* recursive */);
+        } else {
+          // just drop the version-hint.txt file
```

Comment:
The version hint is a hint. If any metadata file exists, then the table will still exist.

I think that the original version is correct. For Hadoop tables, dropping a table means deleting its directory. The confusion here is one reason why Hadoop tables are not recommended for production use.

Reply:
@rdblue: yeah. That's exactly why test cases are failing. 

Also, it is odd to me that `HadoopCatalog` extends `BaseMetastoreCatalog` (as it should not be a metastore catalog. should be file system catalog). 

I will mostly close this PR. 

We are working on a catalog migration API (any catalog to any catalog and of course we will contribute it here). API is simple. But adding testcase with cross catalog is more work. 
While adding API, after catalog migration, we call drop table with purge = false on source catalog. At that time Hadoop catalog was cleaning the migrated table's data too. Hence opened this PR.